util/src/main/java/io/grpc/util/SpiffeIdParserImpl.java

```diff
@@ -0,0 +1,85 @@
+package io.grpc.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class SpiffeIdParserImpl implements SpiffeIdParser {
+
+  private final static String PREFIX = "spiffe://";
+
+  @Override
+  public SpiffeIdInfo parse(String uri) {
+    validateFormat(uri);
+    String domainAndPath = uri.substring(PREFIX.length());
+    String trustDomain;
+    String path = "";
+    if (!domainAndPath.contains("/")) {
+      trustDomain = domainAndPath;
+    } else {
+      String[] parts = domainAndPath.split("/", 2);
+      trustDomain = parts[0];
+      path = parts[1];
+    }
+    validateTrustDomain(trustDomain);
+    validatePath(path);
+    if (!path.isEmpty()) {
+      path = "/" + path;
+    }
+    return new SpiffeIdInfoImpl(trustDomain, path);
+  }
+
+  private static void validateFormat(String uri) throws IllegalArgumentException {
```

Comment:
nit: The method name "validateFormat" is misleading, because are not fully validating the format. I'd suggest renaming to e.g. "doInitialFormatValidation".

Reply:
I did this split to mimic the structure of the [spec](https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/JsonParser.java)

- Spiffe Id
- Trust Domain
- Path

Would it make sense to rename it to `validateSpiffeId` then?

Reply:
Per our offline discussion let's rename to something not misleading

Reply:
renamed to 'doInitialUriValidation'